src/api/route-services/pipeline-response.js

```diff
@@ -78,9 +80,54 @@ const pipelineResponse = async (io, message) => {
     ]);
   }
 
-
   const statusRes = await pipelineStatus(experimentId);
 
+  if (taskName === 'configureEmbedding') {
+    const now = new Date();
```

Comment:
Can we extract this to its own function? And, to ease @ivababukova's concerns, could we explicitly validate the work requests here?

Reply:
Extracted into `hookRunner` and associated functions under `src/utils/hooks`